vm/src/hint_processor/builtin_hint_processor/bootloader_hints.rs

```diff
@@ -0,0 +1,191 @@
+use std::collections::HashMap;
+
+use serde::Deserialize;
+
+use crate::hint_processor::builtin_hint_processor::hint_utils::insert_value_from_var_name;
+use crate::hint_processor::hint_processor_definition::HintReference;
+use crate::serde::deserialize_program::ApTracking;
+use crate::types::exec_scope::ExecutionScopes;
+use crate::vm::errors::hint_errors::HintError;
+use crate::vm::errors::vm_errors::VirtualMachineError;
+use crate::vm::runners::builtin_runner::OutputBuiltinRunner;
+use crate::vm::vm_core::VirtualMachine;
+
+/// Names for internal variables stored in the execution scope.
+mod vars {
+    /// Deserialized bootloader input.
+    pub(crate) const BOOTLOADER_INPUT: &str = "bootloader_input";
+
+    /// Saved state of the output builtin.
+    pub(crate) const OUTPUT_BUILTIN_STATE: &str = "output_builtin_state";
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(transparent)]
+pub struct ProgramHash(pub u64);
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct BootloaderConfig {
+    pub simple_bootloader_program_hash: ProgramHash,
+    pub supported_cairo_verifier_program_hashes: Vec<ProgramHash>,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct PackedOutput {
+    // TODO: missing definitions of PlainPackedOutput, CompositePackedOutput
```

Comment:
FYI, I think `PackedOutput` will also need a `fn elements_for_hash()`

Reply:
Probably, we'll add it when we need it. It will probably end up being an enum instead of a struct as well. We'll figure it out.